.eslintrc

```diff
@@ -0,0 +1,129 @@
+root: true
+
+env:
+    browser: true
+    es6: true
+    jquery: true
+    mocha: true
+    node: true
+
+ecmaFeatures:
+    arrowFunctions: true
+    blockBindings: true
+    classes: true
+    defaultParams: true
+    destructuring: true
+    forOf: true
+    generators: true
+    globalReturn: true
+    jsx: true
+    modules: true
+    restParams: true
+    spread: true
+    templateStrings: true
+
+rules:
+    brace-style: [2, "1tbs"]
+    camelcase: [2, { properties: "always" }]
+    comma-dangle: [2, "never"]
+    comma-spacing: 2
+    comma-style: [2, "last"]
+    complexity: [1, 4]
+    consistent-return: 2
+    curly: [2, "all"]
+    default-case: 2
+    dot-notation: [2, { allowKeywords: true }]
+    eol-last: 2
+    eqeqeq: 2
+    func-style: [2, "declaration"]
+    guard-for-in: 2
+    indent: [2, 2, {"SwitchCase": 1}]
+    key-spacing: [2, { beforeColon: false, afterColon: true }]
+    new-cap: 2
+    new-parens: 2
+    no-alert: 2
+    no-array-constructor: 2
+    no-caller: 2
+    no-cond-assign: [2, "always"]
+    no-console: 2
+    no-constant-condition: 2
+    no-debugger: 2
+    no-delete-var: 2
+    no-dupe-keys: 2
+    no-duplicate-case: 2
+    no-empty: 2
+    no-empty-label: 2
+    no-eval: 2
+    no-ex-assign: 2
+    no-extend-native: 2
+    no-extra-bind: 2
+    no-extra-boolean-cast: 2
+    no-extra-parens: 0
+    no-fallthrough: 2
+    no-floating-decimal: 2
+    no-func-assign: 2
+    no-implied-eval: 2
+    no-inner-declarations: 2
+    no-irregular-whitespace: 2
+    no-iterator: 2
+    no-label-var: 2
+    no-labels: 2
+    no-lone-blocks: 2
+    no-loop-func: 2
+    no-mixed-spaces-and-tabs: [2, false]
+    no-multi-spaces: 2
+    no-multi-str: 2
+    no-native-reassign: 2
+    no-nested-ternary: 2
+    no-new: 2
+    no-new-func: 2
+    no-new-object: 2
+    no-new-wrappers: 2
+    no-obj-calls: 2
+    no-octal: 2
+    no-octal-escape: 2
+    no-proto: 2
+    no-redeclare: 2
+    no-return-assign: 2
+    no-script-url: 2
+    no-sequences: 2
+    no-shadow: 2
+    no-shadow-restricted-names: 2
+    no-spaced-func: 2
+    no-sparse-arrays: 2
+    no-trailing-spaces: 2
+    no-undef: 2
+    no-undef-init: 2
+    no-undefined: 2
+    no-underscore-dangle: 2
+    no-unexpected-multiline: 2
+    no-unreachable: 2
+    no-unused-expressions: 2
+    no-unused-vars: [2, {vars: "all", args: "after-used"}]
+    no-use-before-define: 2
+    no-with: 2
+    quotes: [2, "single"]
+    radix: 2
+    semi: 2
+    semi-spacing: [2, {before: false, after: true}]
+    space-after-keywords: [2, "always"]
+    space-before-blocks: [2, "always"]
+    space-before-function-paren: [2, "never"]
+    space-infix-ops: 2
+    space-return-throw-case: 2
+    space-unary-ops: [2, {words: true, nonwords: true}]
+    spaced-comment: [2, "always", { exceptions: ["-"]}]
+    strict: 0
+    use-isnan: 2
+    valid-jsdoc: [2, { prefer: { "return": "returns"}}]
+    wrap-iife: 2
+    yoda: [2, "never"]
```

Comment:
There are a few more recommended items under [Possible Errors](http://eslint.org/docs/rules/#possible-errors) in the docs that we should probably add, specifically:
* [`no-dupe-args`](http://eslint.org/docs/rules/no-dupe-args)
* [`no-extra-semi`](http://eslint.org/docs/rules/no-extra-semi)
* [`no-invalid-regexp`](http://eslint.org/docs/rules/no-invalid-regexp)
* [`valid-typeof`](http://eslint.org/docs/rules/valid-typeof)